shared/private/header.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "system.h"
+
+namespace ink {
```

Comment:
Should be `ink::internal` since this is not a user facing type.

https://github.com/brwarner/inkcpp/blob/master/CONTRIBUTING.md#namespaces